raiden/network/transport/matrix/transport.py

```diff
@@ -238,6 +243,10 @@ def message_is_in_queue(data: _RetryQueue._MessageData) -> bool:
                     message=msg_data.message,
                     reason="Message was removed from queue",
                 )
+            else:
+                if next(msg_data.expiration_generator):
```

Comment:
How does this fix the problem, to me it looks that the if was just moved around

Reply:
nvm, the `for` loop is over `_message_queue` and not `message_texts`, so it was ignoring the expiration generator.

Reply:
but wait, the `_send_raw` is only called with data from `message_texts`, so it doesn't matter that it is iterating over all the messages.

Reply:
The problem was that before the `message_texts` was constructed unconditionally before this loop (without paying attention if the messages were still in the raiden queues).